WireDomain/Sources/WireDomain/Repositories/UpdateEvents/Protocols/UpdateEventsLocalStoreProtocol.swift

```diff
@@ -38,26 +39,34 @@ public protocol UpdateEventsLocalStoreProtocol {
 
     /// Persists an event envelope locally.
     /// - Parameters:
-    ///     - data: The event envelope payload data.
+    ///     - eventEnvelope: The event envelope to persist.
     ///     - index: The event envelope index.
 
     func persistEventEnvelope(
-        _ data: Data,
+        _ eventEnvelope: UpdateEventEnvelope,
         index: Int64
     ) async throws
 
-    /// Fetches stored event envelope payloads.
+    /// Fetches stored event envelopes.
     /// - parameter limit: A fetch limit.
-    /// - returns: A list of event payloads.
+    /// - returns: A list of event envelopes.
 
-    func fetchStoredEventEnvelopePayloads(
+    func fetchStoredEventEnvelopes(
         limit: UInt
-    ) async throws -> [Data]
+    ) async throws -> [UpdateEventEnvelope]
 
     /// Deletes next pending events locally.
     /// - parameter limit: A fetch limit.
 
     func deleteNextPendingEvents(
         limit: UInt
     ) async throws
+
+    /// Delete the event envelope with the given index.
+    /// - parameter index: The index of the envelope to delete
+
+    func deleteEventEnvelope(
+        atIndex index: Int64
+    ) async throws
```

Comment:
I added this so we can store and delete live events (events that come through the push channel one by one)